### ISSUE TITLE
Drop support for PHP < 7.4?

### ISSUE THREAD

**Issue by Chris53897** [php-http/client-common#242]
Combined usage for PHP 7.1, to 7.3 is at 4 Percent for Version 2.
https://packagist.org/packages/php-http/client-common/php-stats#2

**Comment by xabbuh** [comment]
Would dropping support enable other things? For example, could the code be simplified because some code that is only present because of the support for PHP 7.3 could be removed?

**Comment by dbu** [comment]
yeah not much value in supporting it anymore. on the other hand, i usually only drop old versions when they hinder anything. if we would move to 8.0 or 8.1, we could modernize the code. i don't think dropping 7.1-7.3 allows to modernize anything.

if you want to work on it, i suggest we move to 8.1 directly (people on legacy php can keep using the legacy versions of this library. it is pretty stable, so no big hassle to maintain). and modernize the code where we can with new language features and strict types where possible without BC breaks.

eventually we could do a new major version with BC breaks, to have strict typing everywhere.

**Comment by Chris53897** [comment]
Sounds good. I will work on this. I will set PHP 8.1 as minimum in that case.

**Comment by dbu** [comment]
thanks a lot! and as we just saw in  #239 the legacy support does become cumbersome.

please for now no BC breaks, so that we can have a 2.x version with the cleanups and symfony 7 support.

**Comment by jtojnar** [comment]
`composer install` will now fail as guzzlehttp/psr7 1 is EOL and full of security issues:

```
No composer.lock file present. Updating dependencies to latest instead of installing from lock file. See https://getcomposer.org/install for more information.
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires guzzlehttp/psr7 ^1.4, found guzzlehttp/psr7[1.4.0, ..., 1.9.1] but these were not loaded, because they are affected by security advisories ("PKSA-vznr-tgp9-fd7d", "PKSA-7qs6-zvnz-h66r", "PKSA-gm5x-j3mz-71n9", "PKSA-jj5t-2zs1-dcfm", "PKSA-hn62-zkx4-1y5q", "PKSA-gvzg-s447-b5b5"). Go to https://packagist.org/security-advisories/ to find advisory details. To ignore the advisories, add their IDs to the "policy.advisories.ignore-id" config or add the package to "policy.advisories.ignore". To turn the feature off entirely, you can set "policy.advisories.block" to false.
```

But we cannot just bump it since it does not support PHP 7.1.
